app/controllers/nonsipp/totalvaluequotedshares/TotalValueQuotedSharesCYAController.scala

```diff
@@ -83,79 +134,139 @@ class TotalValueQuotedSharesCYAController @Inject()(
           Redirect(navigator.nextPage(TotalValueQuotedSharesCYAPage(srn), NormalMode, request.userAnswers))
       }
   }
+
+  def onSubmitViewOnly(srn: Srn, year: String, current: Int, previous: Int): Action[AnyContent] =
+    identifyAndRequireData(srn, ViewOnlyMode, year, current, previous).async {
+      Future.successful(
+        Redirect(controllers.nonsipp.routes.ViewOnlyTaskListController.onPageLoad(srn, year, current, previous))
+      )
+    }
 }
 
 object TotalValueQuotedSharesCYAController {
   def viewModel(
     srn: Srn,
     totalCost: Money,
     taxYearOrAccountingPeriods: Either[DateRange, NonEmptyList[(DateRange, Max3)]],
-    schemeDetails: SchemeDetails
-  ): FormPageViewModel[CheckYourAnswersViewModel] =
+    schemeDetails: SchemeDetails,
+    mode: Mode,
+    viewOnlyViewModel: Option[ViewOnlyViewModel]
+  ): FormPageViewModel[CheckYourAnswersViewModel] = {
+    val (title, heading) = mode.fold(
+      normal = ("checkYourAnswers.title", "checkYourAnswers.heading"),
+      check = ("checkYourAnswers.title", "checkYourAnswers.heading"),
+      viewOnly = ("totalValueQuotedSharesCYA.title.view", "totalValueQuotedSharesCYA.heading.view")
+    )
+
     FormPageViewModel[CheckYourAnswersViewModel](
-      title = "checkYourAnswers.title",
-      heading = "checkYourAnswers.heading",
+      title = title,
+      heading = heading,
       description = None,
       page = CheckYourAnswersViewModel(
         sections(
           srn,
           totalCost,
           taxYearOrAccountingPeriods,
           schemeDetails,
-          CheckMode
+          mode
         )
       ),
       refresh = None,
-      buttonText = "site.saveAndContinue",
-      onSubmit = controllers.nonsipp.totalvaluequotedshares.routes.TotalValueQuotedSharesCYAController.onSubmit(srn)
+      buttonText =
+        mode.fold(normal = "site.saveAndContinue", check = "site.continue", viewOnly = "site.return.to.tasklist"),
```

Comment:
According to the content doc, the button text should be "Save and continue" for both `NormalMode` and `CheckMode` - if that's right, maybe this could be amended in the next PR? In any case I'll approve the PR as it wouldn't make sense to hold up merging for something small like this.

Reply:
ah good spot! i'll amend in the next one 👍